File: vbnet/number_guessing_game/number_guessing_game_3.vb
' This program is a number guessing game written in Visual Basic .NET.
' The user has to guess a randomly generated number between 1 and 100.
' The program will provide feedback whether the guess is too high, too low, or correct.

Imports System

Module NumberGuessingGame
    Sub Main()
        Dim random As New Random()
        Dim numberToGuess As Integer = random.Next(1, 101)
        Dim userGuess As Integer = 0
        Dim attempts As Integer = 0
        Dim input As String

        Console.WriteLine("Welcome to the Number Guessing Game!")
        Console.WriteLine("Guess a number between 1 and 100.")

        While userGuess <> numberToGuess
            Console.Write("Enter your guess: ")
            input = Console.ReadLine()
            If Integer.TryParse(input, userGuess) Then
                attempts += 1
                If userGuess < numberToGuess Then
                    Console.WriteLine("Too low! Try again.")
                ElseIf userGuess > numberToGuess Then
                    Console.WriteLine("Too high! Try again.")
                Else
                    Console.WriteLine("Congratulations! You guessed the number in " & attempts & " attempts.")
                End If
            Else
                Console.WriteLine("Invalid input. Please enter a number.")
            End If
        End While

        Dim file As System.IO.StreamWriter
        file = My.Computer.FileSystem.OpenTextFileWriter("game_log.txt", True)
        file.WriteLine("Number to guess: " & numberToGuess & ", Attempts: " & attempts)
    End Sub
End Module

File: csharp/number_guessing_game/number_guessing_game_3.cs
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

/*
 * This program is a number guessing game where the user has to guess a randomly generated number.
 * The program provides feedback on whether the guess is too high or too low.
 * The game continues until the user guesses the correct number.
 */

int main() {
    int number, guess, attempts = 0;
    FILE *logFile;

    srand(time(0));
    number = rand() % 100 + 1;

    logFile = fopen("game_log.txt", "a");

    printf("Welcome to the Number Guessing Game!\n");
    printf("I have selected a number between 1 and 100. Can you guess it?\n");

    do {
        printf("Enter your guess: ");
        scanf("%d", &guess);
        attempts++;

        if (guess > number) {
            fprintf(logFile, "Attempt %d: %d is too high.\n", attempts, guess);
            printf("Too high! Try again.\n");
        } else if (guess < number) {
            fprintf(logFile, "Attempt %d: %d is too low.\n", attempts, guess);
            printf("Too low! Try again.\n");
        } else {
            fprintf(logFile, "Attempt %d: %d is correct!\n", attempts, guess);
            printf("Congratulations! You guessed the number in %d attempts.\n", attempts);
        }
    } while (guess != number);

    return 0;
}